test/integration/host-config.spec.ts

```diff
@@ -415,5 +455,352 @@ describe('host-config', () => {
 		const { network: newNetwork } = await get();
 		expect(network.hostname).to.equal(newNetwork.hostname);
 		expect(network.proxy).to.deep.equal(newNetwork.proxy);
+		await expect(fs.readFile(redsocksConf, 'utf-8')).to.eventually.equal(
+			stripIndent`
+				base {
+					log_debug = off;
+					log_info = on;
+					log = stderr;
+					daemon = off;
+					redirector = iptables;
+				}
+
+				redsocks {
+					ip = example.org;
+					port = 1080;
+					type = socks5;
+					login = "foo";
+					password = "bar";
+					local_ip = 127.0.0.1;
+					local_port = 12345;
+				}
+
+				dnsu2t {
+					remote_ip = 1.2.3.4;
+					remote_port = 54;
+					local_ip = 127.0.0.1;
+					local_port = 53;
+				}` + '\n',
+		);
+	});
+
+	it('patches dnsu2t config to default without modifying other fields', async () => {
+		await patch({
+			network: {
+				proxy: {
+					dns: true,
+				},
+			},
+		});
+		const { network } = await get();
+		expect(network.proxy).to.deep.equal({
+			...defaultConf.proxy,
+			dns: '8.8.8.8:53',
+		});
+		expect(await config.get('hostname')).to.equal(defaultConf.hostname);
+		await expect(fs.readFile(redsocksConf, 'utf-8')).to.eventually.equal(
+			stripIndent`
+				base {
+					log_debug = off;
+					log_info = on;
+					log = stderr;
+					daemon = off;
+					redirector = iptables;
+				}
+
+				redsocks {
+					ip = example.org;
+					port = 1080;
+					type = socks5;
+					login = "foo";
+					password = "bar";
+					local_ip = 127.0.0.1;
+					local_port = 12345;
+				}
+
+				dnsu2t {
+					remote_ip = 8.8.8.8;
+					remote_port = 53;
+					local_ip = 127.0.0.1;
+					local_port = 53;
+				}` + '\n',
+		);
+	});
+
+	it('patches dnsu2t config to string value', async () => {
+		await patch({
+			network: {
+				proxy: {
+					dns: '4.3.2.1:51',
+				},
+			},
+		});
+		const { network } = await get();
+		expect(network.proxy).to.deep.equal({
+			...defaultConf.proxy,
+			dns: '4.3.2.1:51',
+		});
+		await expect(fs.readFile(redsocksConf, 'utf-8')).to.eventually.equal(
+			stripIndent`
+				base {
+					log_debug = off;
+					log_info = on;
+					log = stderr;
+					daemon = off;
+					redirector = iptables;
+				}
+
+				redsocks {
+					ip = example.org;
+					port = 1080;
+					type = socks5;
+					login = "foo";
+					password = "bar";
+					local_ip = 127.0.0.1;
+					local_port = 12345;
+				}
+
+				dnsu2t {
+					remote_ip = 4.3.2.1;
+					remote_port = 51;
+					local_ip = 127.0.0.1;
+					local_port = 53;
+				}` + '\n',
+		);
+	});
+
+	it('patches dnsu2t config with partial string value, using defaults where needed', async () => {
+		await patch({
+			network: {
+				proxy: {
+					// Missing ip
+					dns: ':51',
```

Comment:
Do we really want to support this? I think we can just either a boolean or a `host:port` pair, this would also simplify code

Reply:
Made the dns input type more strict (`t.string` --> `AddressString`), which will fail `hostConfig.parse` if dns is not a string, there is no colon, or there are no characters before or after the colon. Therefore I removed all the test cases that handle missing address / port.